artifactory/commands/python/poetry.go

```diff
@@ -134,6 +144,62 @@ func (pc *PoetryCommand) SetPypiRepoUrlWithCredentials() error {
 	return nil
 }
 
+func ConfigPoetryRepo(url, username, password, configRepoName string) error {
+	err := RunPoetryConfig(url, username, password, configRepoName)
+	if err != nil {
+		return err
+	}
+
+	// Add the repository config to the pyproject.toml
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	if err = addRepoToPyprojectFile(filepath.Join(currentDir, pyproject), configRepoName, url); err != nil {
+		return err
+	}
+	return poetryUpdate()
+}
+
+func RunPoetryConfig(url, username, password, configRepoName string) error {
+	// Add the poetry repository config
+	// poetry config repositories.<repo-name> https://<your-artifactory-url>/artifactory/api/pypi/<repo-name>/simple
+	err := RunConfigCommand(project.Poetry, []string{poetryConfigRepoPrefix + configRepoName, url})
+	if err != nil {
+		return err
+	}
+
+	// Set the poetry repository credentials
+	// poetry config http-basic.<repo-name> <user> <password/token>
+	return RunConfigCommand(project.Poetry, []string{poetryConfigAuthPrefix + configRepoName, username, password})
+}
+
+func poetryUpdate() (err error) {
+	log.Info("Running Poetry update")
+	cmd := gofrogcmd.NewCommand("poetry", "update", []string{})
+	err = gofrogcmd.RunCmd(cmd)
+	if err != nil {
+		return errorutils.CheckErrorf("Poetry config command failed with: %s", err.Error())
+	}
+	return
+}
+
+func addRepoToPyprojectFile(filepath, poetryRepoName, repoUrl string) error {
+	viper.SetConfigType("toml")
+	viper.SetConfigFile(filepath)
+	err := viper.ReadInConfig()
+	if err != nil {
+		return errorutils.CheckErrorf("Failed to read pyproject.toml: %s", err.Error())
+	}
```

Comment:
Let's simplify to - 
```
	if err := viper.ReadInConfig(); err != nil {
		return errorutils.CheckErrorf("Failed to read pyproject.toml: %s", err.Error())
	}
```
Let's do the same for the `err = viper.WriteConfig()` line below.